gneiss/util.py

```diff
@@ -0,0 +1,117 @@
+def match(x, y, intersect=False):
+    """ Sorts samples in metadata and contingency table in the same order.
+
+    Parameters
+    ----------
+    x : pd.DataFrame
+        Contingency table where samples correspond to rows and
+        features correspond to columns.
+    y: pd.DataFrame
+        Metadata table where samples correspond to rows and
+        explanatory metadata variables correspond to columns.
+    intersect : bool, optional
+        Specifies if only the intersection of samples in the
+        contingency table and the metadata table will returned.
+
+    Returns
+    -------
+    _x : pd.DataFrame
+        Filtered dataframe
+    _y : pd.DataFrame
+        Filtered dataframe
+    """
+    _x = x.sort_index()
+    _y = y.sort_index()
+    if intersect:
+        idx = set(_x.index) & set(_y.index)
+        idx = sorted(idx)
+        return _x.loc[idx], _y.loc[idx]
+    else:
+        if len(_x.index) != len(_y.index):
+            raise ValueError("`x` and `y` have incompatible sizes, "
+                             "`x` has %d rows, `y` has %d rows.  "
+                             "Consider setting `intersect=True`." %
+                             (len(_x.index), len(_y.index)))
+        return _x, _y
+
+
+def match_tips(table, tree, intersect=False):
+    """ Returns the OTU table and tree with matched tips.
+
+    Sorts the columns of the OTU table to match the tips in
+    the tree.  If the tree is multi-furcating, then the
+    tree is reduced to a bifurcating tree by randomly inserting
```

Comment:
Probably better to point here to the [skbio `bifurcate`](http://scikit-bio.org/docs/latest/generated/skbio.tree.TreeNode.bifurcate.html#skbio.tree.TreeNode.bifurcate) documentation